### ISSUE TITLE
[BUG] Volumes bounded on a deleted VM and unable to be removed

### ISSUE THREAD

**Issue by lanfon72** [harvester/harvester#2677]
### Describe the bug
https://user-images.githubusercontent.com/5169694/185370936-e07979ed-0079-4754-b2d0-bcb9872a5ebe.mp4

It discovered by daily e2e testing.

### To Reproduce
1. Install Harvester with any nodes
2. Create a Image for VM creation
3. Create the VM `vm1` via API endpoint: `https://<CLUSTER_VIP>/apis/kubevirt.io/v1/namespaces/default/virtualmachines` (POST method)
4. Delete the VM `vm1` and volumes simultaneously. (DELETE method)
    - VM's API endpoint: `https://<CLUSTER_VIP>/apis/kubevirt.io/v1/namespaces/default/virtualmachines/<VM_NAME>`
    - Volume's API endpoint: `https://<CLUSTER_VIP>/api/v1/namespaces/default/persistentvolumeclaims/<VOLUME_NAME>`

### Expected behavior
Volumes should able to be removed

## Environment:
 - Harvester ISO version: **v1.0.3**
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): **Baremetal DL360 4 nodes**

### Additional context
Dashboard's operation will NOT encountered this.

The issue can be reproduced by running `test_create_vm_do_not_start` e2e tests several times.

**Comment by bk201** [comment]
@lanfon72 I guess this is expected since the volume is still used.

If this blocks some QA tests, perhaps it's better to use calls like:

```
DELETE /v1/harvester/kubevirt.io.virtualmachines/default/test-vm-1?removedDisks=disk-0
```

Or add some code to wait for the volume to be unbound.

**Comment by lanfon72** [comment]
@bk201 Volumes attached VM has been removed, but volumes still here and can not be deleted.

this is not always reproduced, and only happened when the VM create and delete simultaneously.

**Comment by Martin-Weiss** [comment]
I have seen this also when using the terraform provider for Harvester to create and then destroy VMs..

**Comment by bk201** [comment]
The owner annotation in volume is not removed. Some investigation:
```
harvester-lzxcs:~ # kubectl get pvc
NAME                                            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                                AGE
025fb39222fd4514a21bf7deeb438af6-disk-0-iok9h   Bound    pvc-50fd54b4-eadb-41b5-a030-8d3f1b938a71   10Gi       RWX            longhorn-2e5b1ca509ce4c96a3570bac0aac2d4d   6d23h
025fb39222fd4514a21bf7deeb438af6-disk-1-4xe1k   Bound    pvc-31218042-2e0b-458a-925e-a95b3328d1ff   10Gi       RWX            longhorn                                    6d23h
5d10e6b574f34bbdb52f00a730c58cd4-disk-0-exno6   Bound    pvc-2e6d0f86-96ff-4fdd-ab5d-ecff10dc56e6   10Gi       RWX            longhorn-15aee8028e514ed5b49d769b0e02f522   6d22h
5d10e6b574f34bbdb52f00a730c58cd4-disk-1-pxrdc   Bound    pvc-264048e8-d662-499e-9cb5-fb47f5e4f7e4   10Gi       RWX            longhorn                                    6d22h
c8220e2392b94b879ee27f5ea0eb0d16-disk-0-3kv8w   Bound    pvc-950e2d3f-b875-409f-bff7-f344abaa6cdf   10Gi       RWX            longhorn-4100e9d817354144833414fbaafc8c50   20h
c8220e2392b94b879ee27f5ea0eb0d16-disk-1-8npwc   Bound    pvc-ba1772a0-7314-4036-976c-ad7c7f3bfa4d   10Gi       RWX            longhorn                                    20h
s1-disk-0-sg2ma                                 Bound    pvc-381d8765-191b-4c93-9f67-3703c767074a   10Gi       RWX            longhorn-image-h4zks                        13d
```

From Harvester log:
```
# grep c8220e harvester-797b59b958-6swcz.log
time="2022-08-24T12:21:46Z" level=error msg="error syncing 'default/c8220e2392b94b879ee27f5ea0eb0d16': handler VMController.UnsetOwnerOfPVCs: Operation cannot be fulfilled on virtualmachines.kubevirt.io \"c8220e2392b94b879ee27f5ea0eb0d16\": StorageError: invalid object, Code: 4, Key: /registry/kubevirt.io/virtualmachines/default/c8220e2392b94b879ee27f5ea0eb0d16, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 93904e3a-e82c-495d-b5b1-65601cd735a2, UID in object meta: , requeuing"
```

QA manually deletes stale volume `pvc-31218042-2e0b-458a-925e-a95b3328d1ff` from LH GUI, stuck in deleting. log in longhorn-manager
```
time="2022-08-25T08:32:02Z" level=warning msg="Dropping Longhorn volume longhorn-system/pvc-31218042-2e0b-458a-925e-a95b3328d1ff out of the queue" controller=longhorn-volume error="fail to sync longhorn-system/pvc-31218042-2e0b-458a-925e-a95b3328d1ff: admission webhook \"validator.harvesterhci.io\" denied the request: can not delete the volume 025fb39222fd4514a21bf7deeb438af6-disk-1-4xe1k which is currently attached to VMs: default/025fb39222fd4514a21bf7deeb438af6" node=harvester-lzxcs
```
[harvester-797b59b958-6swcz.log](https://github.com/harvester/harvester/files/9422959/harvester-797b59b958-6swcz.log)
[supportbundle_2df6b5dc-0664-42d3-85e2-9b151792a654_2022-08-25T08-20-16Z 2.zip](https://github.com/harvester/harvester/files/9422972/supportbundle_2df6b5dc-0664-42d3-85e2-9b151792a654_2022-08-25T08-20-16Z.2.zip) (drop some files to reduce size)

**Comment by bk201** [comment]
This is still reproducible in v1.1.1 master.

**Comment by lanfon72** [comment]
backport PR #3480 not merge yet.

**Comment by ibrokethecloud** [comment]
Can we please test the following case:
* Create a v1.1.1 harvester cluster
* launch a VM
* upgrade cluster to v1.1.2-rc1
* delete the VM
* VM should get deleted

**Comment by lanfon72** [comment]
@ibrokethecloud I think we can just verify this will not happen on `v1.1.2-rc1` (upgraded from `v1.1.1` or new cluster), because it is not always reproducible.

And we can also test the case that reproduced on `v1.1.1` and upgrade to `v1.1.2-rc1`, then volumes should able to be deleted successfully.

**Comment by ibrokethecloud** [comment]
the fixes for last scenario are now available in v1.1-head

**Comment by lanfon72** [comment]
Close as resolved, reproduced on **v1.1-a002d078-head** 100 times, not encountered this.


Test Information
----
* Environment: **qemu/KVM 2 nodes**
* Harvester Version: **v1.1-a002d078-head**
* **ui-source** Option: **Auto**

### Verify Steps:
1. Install Harvester with any nodes
2. Clone https://github.com/harvester/tests#run_tests and setup the test environment and config.yml
3. Execute `pytest harvester_e2e_tests/ -vvvk "test_create_vm_do_not_start"` several times
